services/streammanager/wunderkind/wunderkindmanager_test.go

```diff
@@ -0,0 +1,144 @@
+package wunderkind
+
+import (
+	"encoding/json"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/rudderlabs/rudder-go-kit/logger/mock_logger"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	mock_lambda "github.com/rudderlabs/rudder-server/mocks/services/streammanager/lambda"
+	"github.com/rudderlabs/rudder-server/services/streammanager/common"
+)
+
+var (
+	sampleMessage    = "sample payload"
+	sampleFunction   = "sampleLambdaFunction"
+	sampleExternalID = "sampleExternalID"
+	sampleIAMRoleARN = "sampleRoleArn"
+	invocationType   = "RequestResponse"
+)
+
+func TestNewProducer(t *testing.T) {
+	t.Setenv("WUNDERKIND_REGION", "us-east-1")
+	t.Setenv("WUNDERKIND_IAM_ROLE_ARN", sampleIAMRoleARN)
+	t.Setenv("WUNDERKIND_EXTERNAL_ID", sampleExternalID)
+	t.Setenv("WUNDERKIND_LAMBDA", sampleFunction)
+
+	destinationConfig := map[string]interface{}{}
+	destination := backendconfig.DestinationT{
+		Config:      destinationConfig,
+		WorkspaceID: "sampleWorkspaceID",
+	}
+	timeOut := 10 * time.Second
+	producer, err := NewProducer(&destination, common.Opts{Timeout: timeOut})
+	assert.Nil(t, err)
+	assert.NotNil(t, producer)
+	assert.NotNil(t, producer.client)
```

Comment:
We can probably use require for assertions. Although internally it uses assert only but since we are using require in whole of rudder server. It might be better to use it.
```suggestion
	require.Nil(t, err)
	require.NotNil(t, producer)
	require.NotNil(t, producer.client)
```

Reply:
addressed